app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -119,33 +133,29 @@ def available_subnets
       subnets
     end
 
-    def virtual_networks(region = nil)
-      if region.nil?
-        sdk.vnets
-      else
-        stripped_region = region.gsub(/\s+/, '').downcase
-        sdk.vnets.select { |vnet| vnet.location == stripped_region }
-      end
+    def virtual_networks
+      sdk.vnets.select { |vnet| vnet.location == region }
```

Comment:
Since the region is constant, I think it's worth memoizing the result:

```suggestion
      @virtual_networks ||= sdk.vnets.select { |vnet| vnet.location == region }
```

Reply:
done.